### ISSUE TITLE
How to test your model on my dataset, png format

### ISSUE THREAD

**Issue by stonecropa** [DeweiHu/OCT_DDPM#10]
Hello, I want to use your pre-trained model to denoise my oct data set, but I see that your code is all in nii format, and my images are in png format. I don’t know how to modify it. to test directly on your pre-trained model. I would be grateful if you could answer me.

**Comment by DeweiHu** [comment]
Hi, the image format doesn't matter as long as you convert it to pytorch tensor with shape [b, c, h, w] and the intensity normalized to range [1, 3].

**Comment by stonecropa** [comment]
Thank you.  I know how to modify it, if I have any questions, I will come and ask you. Thank you very much.

**Comment by stonecropa** [comment]
Hello, it may be because of the noise in my image. There is still a certain gap between the clear image generated by the pre-trained model and the clear image in your paper.The first image is the original image, and the others are denoised images.(it is t=40, 45 50,55,60,65 ). Do I need to retrain?

![1](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/6cd7430d-083e-4a88-ab76-0b100225bf95)
![image_40](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/aed5a14a-b987-4848-9c85-ca192c9c91f1)
![image_45](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/14b677af-538b-4cda-abe2-e11942e4cb2b)
![image_5050](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/76b5832f-ac1f-40dd-9799-3a3e3597a79c)
![image_55](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/b5316dfb-d135-44a7-a6e5-2bc8f829fb87)
![image_60](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/7d401820-5b52-4f38-bf51-4450a24345c7)
![image_65](https://github.com/DeweiHu/OCT_DDPM/assets/114655828/aab2673a-8360-46d1-bbaa-2e98193b5376)

**Comment by DeweiHu** [comment]
Yes, to acquire better performance, you probably need to fine-tune the model on your own dataset considering the inherent distribution shift. And you can print the whole spectrum of t to pick the optimal value.
________________________________

**Comment by stonecropa** [comment]
@DeweiHu 
Hello, how can I fine-tune your pre-trained model, and how can I print out the frequency spectrum of t? I would be very grateful if you could tell me the detailed steps.

**Comment by san-kou7** [comment]
> 谢谢。我知道如何修改它，如果我有任何问题，我会来问你。谢谢。

Hello, I would like to know how you modified the code or processed images. My own images are not in nii format. If you could answer me, I would greatly appreciate it @stonecropa

**Comment by gguyinging** [comment]
> 您好，如何微调您的预训练模型，以及如何打印出 t 的频谱？如果您能告诉我详细的步骤，我将不胜感激。
您好，我想知道您是如何修改代码或处理图像的，如果您能回答我，我将不胜感激

**Comment by nathankvphan** [comment]
Hello @stonecropa, I was wondering how you were able to modify the code such that you can use png formatted images. I would greatly appreciate any help you can give.
